src/core/raster/qgsrasterdataprovidertemporalcapabilities.cpp

```diff
@@ -0,0 +1,135 @@
+/***************************************************************************
+                         qgsrasterdataprovidertemporalcapabilities.cpp
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsrasterdataprovidertemporalcapabilities.h"
+
+QgsRasterDataProviderTemporalCapabilities::QgsRasterDataProviderTemporalCapabilities( bool enabled )
+  :  QgsDataProviderTemporalCapabilities( enabled )
+{
+}
+
+//QgsRasterDataProviderTemporalProperties &QgsRasterDataProviderTemporalProperties::operator=( const QgsRasterDataProviderTemporalProperties &other )
+//{
+//  mRange = other.mRange;
+//  mFixedRange = other.mFixedRange;
+//  mEnableTime = other.mEnableTime;
+
+//  return *this;
+//}
+
+void QgsRasterDataProviderTemporalCapabilities::setTemporalRange( const QgsDateTimeRange &dateTimeRange )
+{
+  if ( !isActive() )
+    setIsActive( true );
+
+  // Don't set temporal range outside fixed temporal range limits,
+  // instead set equal to the fixed temporal range
+  QDateTime begin;
+  QDateTime end;
+
+  if ( dateTimeRange.begin() < mFixedRange.begin() )
```

Comment:
Should you check if mFixedRange.isValid() here ? What if a provider doesn't have a fixed range?

Reply:
Yap